src/gitlab/merge_request.py

```
@@ -68,15 +68,54 @@ def get_merge_request_diff(project_id: str, mr_number: str):
+        ):
+            start_index = i
+            break
+
+    # 如果找不到指定的 commit，返回所有 commits
+    if start_index is None:
+        return all_commits
+
+    # 返回从指定 commit 开始到最新的所有 commits（包含起始 commit）
+    # GitLab API 返回的 commits 是倒序的，最新的在前面
+    # 所以从索引 0 到 start_index（包含）就是从最新到起始 commit
+    return all_commits[0 : start_index + 1]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)